core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -854,6 +854,9 @@ public int performBatchRecovery(Iterable<Translog.Operation> operations) {
         if (state != IndexShardState.RECOVERING) {
             throw new IndexShardNotRecoveringException(shardId, state);
         }
+        // We set active because we are now writing operations to the engine; this way, if we go idle after some time and become inactive,
+        // we still give sync'd flush a chance to run:
```

Comment:
Ahh, thanks @bleskes, I'll fix the comment to say "so we can invoke the `onShardInactive` listeners" instead ... and explain that sync'd flush won't actually occur :)